### PR TITLE
1 basic project structure

### PR DESCRIPTION
**Pull request by ondrys123**
Added basic CMake project, submodules, etc.

### CODE REVIEW

**Comment by mdugacek** [comment]
My CMake is not able to find `CMake`s in `sparse/external/Catch2` and `sparse/external/nlohmann_json`.

Error message:
```
CMake Error at cmake/Dependencies.cmake:3 (add_subdirectory):
  The source directory

    /path/to/sparse/external/Catch2

  does not contain a CMakeLists.txt file.
Call Stack (most recent call first):
  CMakeLists.txt:17 (include)


CMake Error at cmake/Dependencies.cmake:6 (add_subdirectory):
  The source directory

    /path/to/sparse/sparse/external/nlohmann_json

  does not contain a CMakeLists.txt file.
Call Stack (most recent call first):
  CMakeLists.txt:17 (include)
```

File: modules/parser/main.cpp [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#include <sparse/common/utility.hpp>
```
Comment (mdugacek) [inline modules/parser/main.cpp:1]: We should probably use `#include "..."` for includes from our project.

File: src/common/utility.cpp [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#include <sparse/common/utility.hpp>
```
Comment (mdugacek) [inline src/common/utility.cpp:1]: Again, we should use `"..."`.

File: src/common/utility.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+#include <iostream>
+#include <nlohmann/json.hpp>
```
Comment (mdugacek) [inline src/common/utility.cpp:4]: I would also use `#include "..."` for this because `nlohmann` is not installed.

File: tests/common/main.cpp [hunk 4]
```diff
@@ -0,0 +1,2 @@
+#define CATCH_CONFIG_MAIN
+#include <catch2/catch.hpp>
```
Comment (mdugacek) [inline tests/common/main.cpp:2]: Again, we should use "...".

File: tests/common/tests_common.cpp [hunk 5]
```diff
@@ -0,0 +1,6 @@
+#include <catch2/catch.hpp>
```
Comment (mdugacek) [inline tests/common/tests_common.cpp:1]: And again, we should use "...".

**Review by mdugacek — CHANGES_REQUESTED** [review]
My CMake is not able to find `CMakeLists` in `sparse/external/Catch2` and `sparse/external/nlohmann_json`.

Error message:
```
CMake Error at cmake/Dependencies.cmake:3 (add_subdirectory):
  The source directory

    /path/to/sparse/external/Catch2

  does not contain a CMakeLists.txt file.
Call Stack (most recent call first):
  CMakeLists.txt:17 (include)


CMake Error at cmake/Dependencies.cmake:6 (add_subdirectory):
  The source directory

    /path/to/sparse/sparse/external/nlohmann_json

  does not contain a CMakeLists.txt file.
Call Stack (most recent call first):
  CMakeLists.txt:17 (include)
```

**Review by mdugacek — COMMENTED** [review]
It would be nice to add to `README`  that you have to run
```
$ git submodule init
$ git submodule update
```
before running `cmake`.

**Review by misoxxx — CHANGES_REQUESTED** [review]
V pohode, jedine co mi este chyba je ze ked to skompilujem tak je tam kopa suborov ktore nie su v .gitignore:

        CMakeCache.txt
        CMakeDoxyfile.in
        CMakeDoxygenDefaults.cmake
        CMakeFiles/
        CTestTestfile.cmake
        DartConfiguration.tcl
        Makefile
        _deps/
        cmake_install.cmake
        doc/CMakeFiles/
        doc/CTestTestfile.cmake
        doc/Makefile
        doc/cmake_install.cmake
        doc/html/
        doc/latex/
        modules/CMakeFiles/
        modules/CTestTestfile.cmake
        modules/Makefile
        modules/cmake_install.cmake
        modules/parser/CMakeFiles/
        modules/parser/CTestTestfile.cmake
        modules/parser/Makefile
        modules/parser/cmake_install.cmake
        modules/parser/parser
        src/CMakeFiles/
        src/CTestTestfile.cmake
        src/Makefile
        src/cmake_install.cmake
        src/common/CMakeFiles/
        src/common/CTestTestfile.cmake
        src/common/Makefile
        src/common/cmake_install.cmake
        tests/CMakeFiles/
        tests/CTestTestfile.cmake
        tests/Makefile
        tests/cmake_install.cmake
        tests/common/CMakeFiles/
        tests/common/CTestTestfile.cmake
        tests/common/Makefile
        tests/common/cmake_install.cmake
        tests/common/tests_common

**Pull request merged**